app/models/player.rb

```diff
@@ -2,8 +2,55 @@ class Player < ApplicationRecord
   include Common_Processing
   mount_uploader :image, PlayerUploader
   belongs_to :spike
+  has_many :player_position_tags, dependent: :destroy
+  has_many :player_genre_tags, dependent: :destroy
+  has_many :player_strength_tags, dependent: :destroy
+  has_many :position_tags, through: :player_position_tags
+  has_many :genre_tags, through: :player_genre_tags
+  has_many :strength_tags, through: :player_strength_tags
 
   validates :name,
             :image,
             presence: true
+
+  def save_tags_for_player(position_tags, genre_tags, strength_tags)
+  # ----ポジションタグの処理
+    current_tags = self.position_tags.pluck(:position) unless self.position_tags.nil?
+    old_tags = current_tags - position_tags.split(',')
+    new_tags = position_tags.split(',').uniq - current_tags
+
+    old_tags.each do |old_tag|
+      self.position_tags.delete PositionTag.find_by(position: old_tag)
+    end
+
+    new_tags.each do |new_tag|
+      player_tag = PositionTag.find_or_initialize_by(position: new_tag)
+      self.position_tags.push(player_tag)
+    end
```

Comment:
```suggestion
self.position_tags = position_tag_names.map { |name| PositionTag.find_or_initialize_by(name: name) }
save!
```
「現在の」とか「古い」とか意識せず丸っと上書きした方が良いと思う。

Reply:
position_tags, genre_tags, strength_tags登録用のメソッドをそれぞれ作ってそれをこのメソッドの中から呼び出す方が責務が分かれて良いと思います。

Reply:
あとこの三つのタグの中でpositionだけ性質が違う？ユーザーが自由に設定できるものではなくすでにあるものの中から選ぶ感じになると思ったんだけどどうだろう？

Reply:
![image](https://user-images.githubusercontent.com/52743869/82585306-c216d300-9bd0-11ea-99fa-b08bb07b0f22.png)

[map]がサポートされておらず

Reply:
一旦配列に入れてから処理することでmapを使用できるようにした
46b498c で修正。